be/src/exprs/array_functions.cpp

```diff
@@ -94,14 +94,12 @@ static StatusOr<ColumnPtr> do_array_append(const Column& elements, const UInt32C
 
 // FIXME: A proof-of-concept implementation with poor performance.
 StatusOr<ColumnPtr> ArrayFunctions::array_append([[maybe_unused]] FunctionContext* context, const Columns& columns) {
-    const Column* arg0 = columns[0].get();
-    const Column* arg1 = columns[1].get();
-    if (arg0->only_null()) {
+    if (columns[0]->only_null()) {
         return columns[0];
     }
 
-    arg0 = arg0->has_null() || arg0->is_constant() ? arg0 : ColumnHelper::get_data_column(arg0);
-    arg1 = arg1->has_null() || arg1->is_constant() ? arg1 : ColumnHelper::get_data_column(arg1);
+    const Column* arg0 = ColumnHelper::unpack_and_duplicate_const_column(columns[0]->size(), columns[0]).get();
```

Comment:
suggest to implement another branch for constant columns, reference to substr

Reply:
there are many such cases, we can optimize it later. This PR just avoid crash.